www/scripts/codecheckerviewer/ListOfRuns.js

```diff
@@ -249,6 +250,41 @@ function (declare, dom, ItemFileWriteStore, topic, Dialog, Button,
           }, 500);
         }
       });
+      this.addChild(this._runFilter);
+
+      //--- Delete Button ---//
+
+      if (this.get('showDelete')) {
+        this._deleteBtn = new Button({
+          label    : 'Delete',
+          class    : 'del-btn',
+          disabled : true,
+          onClick  : function () {
+            that.standBy.show();
+
+            CC_SERVICE.removeRunResults(that.deleteRunIds, function (success) {
+              that.listOfRunsGrid.store.fetch({
+                onComplete : function (runs) {
+                  that.standBy.hide();
+
+                  var runCount = runs.length;
+                  runs.forEach(function (run) {
+                    if (that.deleteRunIds.indexOf(run.runid[0]) !== -1) {
+                      that.listOfRunsGrid.store.deleteItem(run);
```

Comment:
I have a concern about this line. Imagine the following scenario:

 1. Select multiple runs for deletion and press `Delete`.
 2. The run IDs selected, as a list, is sent to the server.
 3. The server checks every run manually if it can be deleted (locks and such), and then starts deleting the runs one-by-one, calling a `delete` and a `commit` on the database.
 4. One of the delete fails. In this case an exception is sent back to the client.

What happens in this case? If the callback function doesn't execute, no runs are removed from the table, but potentially clicking on a run that **was** deleted from the database something weird must happen, and a page reload must be done to "fix" the state of the GUI.

Reply:
If the deletion of at least one run fails only False is returned no exception:
https://github.com/Ericsson/codechecker/blob/b9987109e6dac5aff4babf9bac47a34d3ad539e4/libcodechecker/server/api/report_server.py#L1752

We could improve that to give a better return value in case of failure. Maybe a list of run ids would be better where the deletion failed.

Reply:
Good, I don't want an API break either. But the GUI could query all the runs that were attempted to be deleted in case of failure, and only remove the rows from the View which went missing from the server.